samtranslator/model/sam_resources.py

```diff
@@ -290,6 +290,13 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def]
                         self.logical_id,
                         "AutoPublishCodeSha256 must be a string",
                     )
+                # Lambda doesn't create a new version if the code in the unpublished version is the same as the
+                # previous published version. To handle scenario users only update content in 'CodeUri', CloudFormation
+                # won't be able to detect any change in lambda function in the template, which would cause deployment
+                # failure. Append to description to handle this issue
+                description = intrinsics_resolver.resolve_parameter_refs(self.Description)
```

Comment:
What if `Description` uses other intrinsics which we fail to resolve?
Would it be safer to use `!Join` intrinsic to add `code_sha_256`?

Reply:
Sorry my description is not accurate. I meant it will cause an update to users who use CodeSha256 in their template even when there's no change made.

Reply:
If `Description` uses other intrinsics, we would not resolve it here. `!Join` is definitely a viable solution. The reason I didn't use `!Join` is that I feel like it's an overkill since not many users would use `Description` with other intrinsics and use `AutoPublishCodeSha256` at the same time. What do you think?

Reply:
I agree that it sounds like an overkill and in 99.90% there will be no need in `Join` but I was surprised when I got a ticket from users using intrinsics on Name in GraphQL to override LogicalId. It'd say it's better to be safe.